pro-planner-client/src/components/OutingDay.jsx

```diff
@@ -1,160 +1,207 @@
-        const inDecidedSelect = (segment) => decisionPreview[0] <= segment && segment < decisionPreview[1];
-
-        const handleClick = () => {
-            if (isEditing || isDeciding) {
-                onSegmentClick(segmentStart, isSelected);
-            } else {
-                if (isSummaryDate) {
-                    dispatch(setDetailedDay(null));
-                    dispatch(setDetailedUsers([]));
-                } else {
-                    dispatch(setDetailedDay(makeOutingDate(segmentStart)));
-                    dispatch(setDetailedUsers([...isSelected]));
-                }
-            }
-        }
-
-        let step;
-        const ratio = isSelected.length / maxUsers;
-        step = STEP_ARR.reduce((acc, [lower, upper], index) => {
-            if (acc === null) {
-                if (ratio >= lower && ratio <= upper) {
-                    return index + 1;
-                }
-            }
-            return acc;
-        }, null);
-        const segmentLogic = () => {
-            if (isEditing) {
-                return assembleClass(
-                    'available',
-                    inSelection ? 'editing' : isSelectedSegment && `outing-edit-select`,
-                );
-            } else if (isDeciding) {
-                return assembleClass(
-                    'available',
-                    inSelection ? 'deciding-selection' : isSelectedSegment && `outing-selected-${step}`,
-                    decisionPreview && inDecidedSelect(segmentStart) && 'outing-decided-4',
-
-                );
-            } else {
-                return assembleClass(
-                    'available',
-                    isSummaryDate && 'summary',
-                    isSelectedSegment && `outing-selected-${step}`,
-                    isDecision && (inDecidedRange(segmentStart) && `outing-decided-${step}`),
-                );
-            }
-        }
-
-        return <div
-            key={`segment-${format(segmentStart, 'dd-HH-mm')}`}
-            className={`${segmentClass} ${segmentLogic()}`}
-            onClick={handleClick}
-            onMouseEnter={(isEditing || isDeciding) ? () => onSegmentEnter(getEndOfSegment(segmentStart)) : null}
-        />
-    });
-        
-    return <div className="d-flex flex-column day-container">
-        {displaySegments.map((segment) => {
-            return segment
-        })}
-    </div>
+	const dispatch = useDispatch();
+	const decision = useSelector(state => state.planParameters.decisionRange).map(
+		outingString => new Date(outingString)
+	);
+	const summaryState = useSelector(state => state.summary.detailedDay);
+	const summaryDate = summaryState ? new Date(summaryState) : null;
+	const isDecision = !!decision.length;
+
+	// responsible for filtering based on user selected in the sidebar.
+	if (selectedUser) {
+		let result = [];
+		console.log(selections);
+		if (selections) {
+			for (let selection of selections) {
```

Comment:
`selections` should be an object of structure `{<username>: <arr of selections>, ...}`. I don't think a for...of... loop will work for iteration over this.